### ISSUE TITLE
Missing Filter for EinheitBetriebsstatus in power_plants_status_quo Function

### ISSUE THREAD

**Issue by khelfen** [openego/powerd-data#221]
### Issue Description
In the [`power_plants_status_quo`](https://github.com/openego/powerd-data/blob/bad9f704bce388abaf0701401c17724d182717e4/src/egon/data/datasets/power_plants/__init__.py#L907-L1229) function, it appears that there's a missing filter for the `EinheitBetriebsstatus` attribute to ensure that only power plants with a status of "InBetrieb" (In Operation) are represented in the model. Currently, decommissioned conventional power plants are also included, which leads to inaccuracies in the energy model's outcomes.

### Expected Behavior
The function should include a filter that selects only the power plants with an `EinheitBetriebsstatus` of "InBetrieb" to ensure that the model accurately represents operational power plants only.

### Current Behavior
Decommissioned conventional power plants are represented in the model due to the absence of a filter for the `EinheitBetriebsstatus` attribute in the `power_plants_status_quo` function.

### Possible Solution
Implement a filter in the `power_plants_status_quo` function that checks the `EinheitBetriebsstatus` attribute of each power plant and includes only those with a status of "InBetrieb" in the model.

**Comment by khelfen** [comment]
See [8183ecc](https://github.com/openego/powerd-data/commit/8183ecc7fd1e1db089e9e4cd7f29c877a1b4a9d9) for a possible solution.

**Comment by khelfen** [comment]
@ClaraBuettner or @CarlosEpia besides conventional power plants other power plants are loaded from the DB (e.g. [hydro](https://github.com/openego/powerd-data/blob/bad9f704bce388abaf0701401c17724d182717e4/src/egon/data/datasets/power_plants/__init__.py#L1095-L1100)). These are also not filtered for `EinheitBetriebsstatus`. Are these filtered before they are written in the DB table (e.g. `supply.egon_power_plants_hydro`) which they are read from?

**Comment by CarlosEpia** [comment]
@khelfen I agree with your observation. I will take care of this issue.

**Comment by CarlosEpia** [comment]
There are two attributes that should be considered here:
1. EinheitBetriebsstatus (Power plant status): currently applied just to not conventional generators [here](https://github.com/openego/powerd-data/blob/bad9f704bce388abaf0701401c17724d182717e4/src/egon/data/datasets/power_plants/mastr.py#L327-L334)
2. DatumEndgueltigeStilllegung (decommission date): Not used yet. I will use this commit 8183ecc as a reference.

**Comment by CarlosEpia** [comment]
Additionally, I will delete this [filter](https://github.com/openego/powerd-data/blob/bad9f704bce388abaf0701401c17724d182717e4/src/egon/data/datasets/power_plants/mastr.py#L316-L325). This will allow us to calculate several scenarios (still a pending decision) and filter the generators according to the requirements of each of them.

**Comment by CarlosEpia** [comment]
@khelfen After implementing the changes I concluded that using the attribute "EinheitBetriebsstatus" == "InBetrieb" is something that we don't want in general. The value presented there corresponds to the current status of the power plant by the moment the data is downloaded. For instance, for the Status2019 we would lose all the generators that were decommissioned after 2020, since they are marked as "DauerhaftStillgelegt".

In conclusion, I think that using the decommissioned date is enough to guarantee the accuracy of the data.

**Comment by khelfen** [comment]
> @khelfen After implementing the changes I concluded that using the attribute "EinheitBetriebsstatus" == "InBetrieb" is something that we don't want in general. The value presented there corresponds to the current status of the power plant by the moment the data is downloaded. For instance, for the Status2019 we would lose all the generators that were decommissioned after 2020, since they are marked as "DauerhaftStillgelegt".
> 
> In conclusion, I think that using the decommissioned date is enough to guarantee the accuracy of the data.

Hey @CarlosEpia :) I'm not sure if I would trust the MaStR to that degree. Did you check if all plants that are "DauerhaftStillgelegt" have a date `DatumEndgueltigeStilllegung`? I did not, but I would suspect that it is not the case. That's why I added [this part](https://github.com/openego/powerd-data/commit/8183ecc7fd1e1db089e9e4cd7f29c877a1b4a9d9#diff-4106960e7e0e57d51c4aca1cb785dafc41cc46b902c4a8ea260d05d964c938c2R1025-R1031) which should work as desired in your case.

**Comment by khelfen** [comment]
If you implement it for all MaStR data could you also update the filter for pv rooftop [here](https://github.com/openego/powerd-data/blob/bad9f704bce388abaf0701401c17724d182717e4/src/egon/data/datasets/power_plants/pv_rooftop_buildings.py#L277-L281)?

**Comment by CarlosEpia** [comment]
Sure. I could also apply it there. Just in case I will mention that we are not currently using that code for "Status20XX" scenarios. We are dealing with solar rooftop [here](https://github.com/openego/powerd-data/blob/bad9f704bce388abaf0701401c17724d182717e4/src/egon/data/datasets/power_plants/__init__.py#L1159-L1195)
